cloudify_agent/api/pm/base.py

```diff
@@ -300,6 +329,37 @@ def validate_optional(self):
         self._validate_autoscale()
         self._validate_host()
 
+    def _get_broker_port(self):
+        """
+        Determines the broker port if it has not been provided. Only intended
+        to be called before self.broker_port has been set and after
+        self.broker_ssl_cert has been set.
+        """
+        if self.broker_ssl_cert == '':
```

Comment:
might be nicer to add the ports to https://github.com/cloudify-cosmo/cloudify-agent/blob/master/cloudify_agent/api/defaults.py: so that they're available for both SSL and non-SSL..
and then do something like this:

```python
return defaults.BROKER_SSL_PORT if self.broker_ssl_cert else defaults.BROKER_PORT
```